test/unit/test/tmp_filename_test.cpp

```diff
@@ -67,13 +67,25 @@ TEST(tmp_filename_dtr, dtr)
     EXPECT_FALSE(fs::exists(path.parent_path()));
 }
 
-// throw if invalid TMPDIR
+// throw if TMPDIR not writeable
```

Comment:
I would need to wrap this inside
```cpp
#if __has_include(<filesystem>)
// Setting the permissions with perm_options is not available in the experimental/filesystem branch.
```

In this case, I would mark the corresponding tested throw inside the `tmp_filename` as unreachable until we use at least gcc8 for coverage?

Reply:
First of all, I find the new version definitely better!

But how do we solve the gcc7 fiasco?

```c++
// pre C++17 with filesystem TS
// see https://en.cppreference.com/w/cpp/experimental/fs/permissions
fs::permissions("test.txt", fs::perms::add_perms | fs::perms::owner_all | fs::perms::group_all);

// real C++17 with final filesystem API
// see https://en.cppreference.com/w/cpp/filesystem/permissions
fs::permissions("test.txt", fs::perms::owner_all | fs::perms::group_all, fs::perm_options::add);
```

Since the parameter changed from 2 to 3 inputs, we could add a new function `std::filesystem::permission` and `std::filesystem::perm_options` that supports this functionality.

Reply:
I added the functionality